controllers/generic_network_controller.go

```diff
@@ -189,6 +190,20 @@ func (r *genericNetworkReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		}
 	} else {
 		reqLogger.Info("NetworkAttachmentDefinition CR already exist")
+
+		foundOwner := found.GetAnnotations()[consts.OwnerRefAnnotation]
+		expectedOwner := netAttDef.GetAnnotations()[consts.OwnerRefAnnotation]
+
+		// Note for the future: the `foundOwner != ""` condition can be removed to make the operator not touching the NetworkAttachmentDefinition created
+		// by the user.
+		if foundOwner != "" && foundOwner != expectedOwner {
```

Comment:
just a quick question if we have `!equality.Semantic.DeepEqual(found.GetAnnotations(), netAttDef.GetAnnotations()` below do we still need this validation here?

Reply:
yes, we do:

in case we have:
```
kind: NetworkAttachmentDefinition
metadata:
  name: foo
  namespace: foo-ns
  annotations:
     owner-ref: SriovNetwork/sriov-operator-ns/foo
spec:
  <sriov cni config>
  ```
which comes from an SriovNetwork in the sriov-operator-ns namespace


when reconciling another network object:
```
kind: SriovNetwork
metadata:
  name: foo
  namespace: foo-ns
spec: ...
```

the generated net-attach-def has `owner-ref: SriovNetwork/foo-ns/foo`, which would make the condition  `!equality.Semantic.DeepEqual(found.GetAnnotations(), netAttDef.GetAnnotations()` true and the netAttachDef woudl be overridden.

The policy, as of now, is the first reconciler networkobject is the owner